### ISSUE TITLE
Exception "Cannot set property 'textContent' of undefined" from a custom block after blockly update 

### ISSUE THREAD

**Issue by jogibear9988** [google/blockly#2214]
### Problem statement

After a Blockly update following Exception is raised:
    
      Uncaught TypeError: Cannot set property 'textContent' of undefined

I use this Code wich worked before

	'use strict';

	goog.provide('Blockly.JavaScript.Sendto');

	goog.require('Blockly.JavaScript');

	Blockly.CustomBlocks = Blockly.CustomBlocks || [];
	Blockly.CustomBlocks.push('Sendto');

	Blockly.Sendto = {
		HUE: 310,
		blocks: {}
	};

	// --- sendTo Custom --------------------------------------------------
	Blockly.Sendto.blocks['sendto_custom'] =
		'<block type="sendto_custom">'
		+ '     <value name="INSTANCE">'
		+ '     </value>'
		+ '     <value name="COMMAND">'
		+ '     </value>'
		+ '     <value name="LOG">'
		+ '     </value>'
		+ '     <value name="WITH_STATEMENT">'
		+ '     </value>'
		+ '     <mutation with_statement="false" items="parameter1"></mutation>'
		+ '</block>';

	Blockly.Blocks['sendto_custom_container'] = {
		/**
		 * Mutator block for container.
		 * @this Blockly.Block
		 */
		init: function () {
			this.setColour(Blockly.Sendto.HUE);

			this.appendDummyInput()
				.appendField(Blockly.Words['sendto_custom_arguments'][systemLang]);

			this.appendStatementInput('STACK');
			this.setTooltip(Blockly.Words['sendto_custom_arg_tooltip'][systemLang]);
			this.contextMenu = false;
		}
	};

	Blockly.Blocks['sendto_custom_item'] = {
		/**
		 * Mutator block for add items.
		 * @this Blockly.Block
		 */
		init: function () {
			this.setColour(Blockly.Sendto.HUE);

			this.appendDummyInput('NAME')
				.appendField(Blockly.Words['sendto_custom_argument'][systemLang]);

			this.setPreviousStatement(true);
			this.setNextStatement(true);
			this.setTooltip(Blockly.Words['sendto_custom_arg_tooltip'][systemLang]);
			this.contextMenu = false;
		}
	};

	Blockly.Blocks['sendto_custom'] = {
		/**
		 * Block for creating a string made up of any number of elements of any type.
		 * @this Blockly.Block
		 */
		init: function () {
			var options = [];
			if (typeof main !== 'undefined' && main.instances) {
				for (var i = 0; i < main.instances.length; i++) {
					if (main.objects[main.instances[i]].common.messagebox) {
						var id = main.instances[i].substring('system.adapter.'.length);
						options.push([id, id]);
					}
				}
				if (!options.length) {
					options.push([Blockly.Words['sendto_no_instances'][systemLang], '']);
				}
				/*for (var h = 0; h < scripts.hosts.length; h++) {
					options.push([scripts.hosts[h], scripts.hosts[h]]);
				}*/
				this.appendDummyInput('INSTANCE')
					.appendField(Blockly.Words['sendto_custom'][systemLang])
					.appendField(new Blockly.FieldDropdown(options), 'INSTANCE');
			} else {
				this.appendDummyInput('INSTANCE')
					.appendField(Blockly.Words['sendto_custom'][systemLang])
					.appendField(new Blockly.FieldTextInput('adapter.0'), 'INSTANCE');
			}

			this.appendDummyInput('COMMAND')
				.appendField(Blockly.Words['sendto_custom_command'][systemLang])
				.appendField(new Blockly.FieldTextInput('send'), 'COMMAND');

			this.setColour(Blockly.Sendto.HUE);

			this.itemCount_ = 1;
			this.updateShape_();
			this.setInputsInline(false);
			this.setPreviousStatement(true, null);
			this.setNextStatement(true, null);
			this.setMutator(new Blockly.Mutator(['sendto_custom_item']));
			this.setTooltip(Blockly.Words['sendto_custom_tooltip'][systemLang]);
			this.setHelpUrl(getHelp('sendto_custom_help'));
		},
		/**
		 * Create XML to represent number of text inputs.
		 * @return {!Element} XML storage element.
		 * @this Blockly.Block
		 */
		mutationToDom: function () {
			var container = document.createElement('mutation');
			var names = [];
			for (var i = 0; i < this.itemCount_; i++) {
				var input = this.getInput('ARG' + i);
				names[i] = input.fieldRow[0].getValue();
			}

			container.setAttribute('items', names.join(','));
			container.setAttribute('with_statement', this.getFieldValue('WITH_STATEMENT') === 'TRUE');
			return container;
		},
		/**
		 * Parse XML to restore the text inputs.
		 * @param {!Element} xmlElement XML storage element.
		 * @this Blockly.Block
		 */
		domToMutation: function (xmlElement) {
			var names = xmlElement.getAttribute('items').split(',');
			this.itemCount_ = names.length;
			this.updateShape_(names, xmlElement.getAttribute('with_statement') == 'true');
		},
		/**
		 * Populate the mutator's dialog with this block's components.
		 * @param {!Blockly.Workspace} workspace Mutator's workspace.
		 * @return {!Blockly.Block} Root block in mutator.
		 * @this Blockly.Block
		 */
		decompose: function (workspace) {
			var containerBlock = workspace.newBlock('sendto_custom_container');
			containerBlock.initSvg();
			var connection = containerBlock.getInput('STACK').connection;
			for (var i = 0; i < this.itemCount_; i++) {
				var itemBlock = workspace.newBlock('sendto_custom_item');
				itemBlock.initSvg();
				connection.connect(itemBlock.previousConnection);
				connection = itemBlock.nextConnection;
			}
			return containerBlock;
		},
		/**
		 * Reconfigure this block based on the mutator dialog's components.
		 * @param {!Blockly.Block} containerBlock Root block in mutator.
		 * @this Blockly.Block
		 */
		compose: function (containerBlock) {
			var itemBlock = containerBlock.getInputTargetBlock('STACK');
			// Count number of inputs.
			var connections = [];
			var names = [];
			while (itemBlock) {
				connections.push(itemBlock.valueConnection_);
				itemBlock = itemBlock.nextConnection &&
					itemBlock.nextConnection.targetBlock();
			}
			// Disconnect any children that don't belong.
			for (var i = 0; i < this.itemCount_; i++) {
				var input = this.getInput('ARG' + i);
				var connection = input.connection.targetConnection;
				names[i] = input.fieldRow[0].getValue();
				if (connection && connections.indexOf(connection) === -1) {
					connection.disconnect();
				}
			}
			this.itemCount_ = connections.length;
			if (this.itemCount_ < 1) this.itemCount_ = 1;
			this.updateShape_(names);
			// Reconnect any child blocks.
			for (var j = 0; j < this.itemCount_; j++) {
				Blockly.Mutator.reconnect(connections[j], this, 'ARG' + j);

			}
		},
		getArgNames_: function () {
			var names = [];
			for (var n = 0; n < this.itemCount_; n++) {
				var input = this.getInput('ARG' + n);
				names.push(input.fieldRow[0].getValue());
			}
			return names;
		},
		/**
		 * Store pointers to any connected child blocks.
		 * @param {!Blockly.Block} containerBlock Root block in mutator.
		 * @this Blockly.Block
		 */
		saveConnections: function (containerBlock) {
			var itemBlock = containerBlock.getInputTargetBlock('STACK');
			var i = 0;
			while (itemBlock) {
				var input = this.getInput('ARG' + i);
				itemBlock.valueConnection_ = input && input.connection.targetConnection;
				itemBlock = itemBlock.nextConnection && itemBlock.nextConnection.targetBlock();
				i++;
			}
		},
		/**
		 * Modify this block to have the correct number of inputs.
		 * @private
		 * @this Blockly.Block
		 */
		updateShape_: function (names, withStatement) {
			if (this.getInput('LOG'))
				this.removeInput('LOG');
			if (this.getInput('WITH_STATEMENT'))
				this.removeInput('WITH_STATEMENT');
			names = names || [];
			var _input;
			// Add new inputs.
			for (var i = 0; i < this.itemCount_; i++) {
				_input = this.getInput('ARG' + i);
				if (!_input) {
					_input = this.appendValueInput('ARG' + i);
					if (!names[i]) names[i] = Blockly.Words['sendto_custom_argument'][systemLang] + (i + 1);
					_input.appendField(new Blockly.FieldTextInput(names[i]));

					var _shadow = this.workspace.newBlock('text');
					_shadow.setShadow(true);
					_shadow.outputConnection.connect(_input.connection);
					_shadow.initSvg();
					_shadow.render();
					//console.log('New ' + names[i]);
				} else {
					_input.fieldRow[0].setValue(names[i]);
					//console.log('Exist ' + names[i]);
					if (!_input.connection.isConnected()) {
						console.log('Create ' + names[i]);
						var shadow = this.workspace.newBlock('text');

						shadow.setShadow(true);

						shadow.outputConnection.connect(_input.connection);
						shadow.initSvg();
						shadow.render();
					}
				}
			}
			// Remove deleted inputs.
			var blocks = [];
			while (_input = this.getInput('ARG' + i)) {
				var b = _input.connection.targetBlock();
				if (b && b.isShadow()) {
					blocks.push(b);
				}
				this.removeInput('ARG' + i);
				i++;
			}
			if (blocks.length) {
				var ws = this.workspace;
				setTimeout(function () {
					for(var b = 0; b < blocks.length; b++) {
						ws.removeTopBlock(blocks[b]);
					}
				}, 100);
			}

			this.appendDummyInput('WITH_STATEMENT')
				.appendField(Blockly.Words['request_statement'][systemLang])
				.appendField(new Blockly.FieldCheckbox(withStatement ? 'TRUE': 'FALSE', function (option) {
					var withStatement = (option == true);
					this.sourceBlock_.updateShape_(this.sourceBlock_.getArgNames_(), withStatement);
				}), 'WITH_STATEMENT');

			this.appendDummyInput('LOG')
				.appendField(Blockly.Words['sendto_log'][systemLang])
				.appendField(new Blockly.FieldDropdown([
					[Blockly.Words['sendto_log_none'][systemLang],  ''],
					[Blockly.Words['sendto_log_info'][systemLang],  'log'],
					[Blockly.Words['sendto_log_debug'][systemLang], 'debug'],
					[Blockly.Words['sendto_log_warn'][systemLang],  'warn'],
					[Blockly.Words['sendto_log_error'][systemLang], 'error']
				]), 'LOG');

			// Add or remove a statement Input.
			var inputExists = this.getInput('STATEMENT');

			if (withStatement) {
				if (!inputExists) {
					this.appendStatementInput('STATEMENT');
				}
			} else if (inputExists) {
				this.removeInput('STATEMENT');
			}
		}
	};

	Blockly.JavaScript['sendto_custom'] = function (block) {
		var instance      = block.getFieldValue('INSTANCE');
		var logLevel      = block.getFieldValue('LOG');
		var command       = block.getFieldValue('COMMAND');
		var withStatement = block.getFieldValue('WITH_STATEMENT');
		var args = [];
		var logText;
		if (logLevel) {
			logText = 'console.' + logLevel + '("' + instance + ': " + "' + (args.length ? args.join(',') + '\n' : '') + '");\n'
		} else {
			logText = '';
		}
		var statement;
		if (withStatement === 'TRUE') {
			statement = Blockly.JavaScript.statementToCode(block, 'STATEMENT');
		}

		for (var n = 0; n < block.itemCount_; n++) {
			var input = this.getInput('ARG' + n);
			var val = Blockly.JavaScript.valueToCode(block, 'ARG' + n, Blockly.JavaScript.ORDER_COMMA);
			// if JSON
			if (val && val[0] === "'" && val[1] === '{') {
				val = val.substring(1, val.length - 1);
			}
			args.push('\n   "' + input.fieldRow[0].getValue() + '": ' + val);

			if (block.itemCount_ === 1 && !input.fieldRow[0].getValue()) {
				if (statement) {
					return 'sendTo("' + instance + '", "' + command + '", ' + val + ', function (result) {\n  ' + statement + '  });\n' + logText;
				} else {
					return 'sendTo("' + instance + '", "' + command + '", ' + val + ');\n' + logText;
				}
			}
		}

		if (statement) {
			return 'sendTo("' + instance + '", "' + command + '", {' + (args.length ? args.join(',') + '\n' : '') + '}, function (result) {\n  ' + statement + '  });\n' + logText;
		} else {
			return 'sendTo("' + instance + '", "' + command + '", {' + (args.length ? args.join(',') + '\n' : '') + '});\n' + logText;
		}
	};

### Expected Behavior

     The custom block should work

### Actual Behavior

    The Exception is raised, in the line 	_shadow.render();

### Steps to Reproduce

use our custom block

### Stack Traces

	blockly_compressed.js:1360 Uncaught TypeError: Cannot set property 'textContent' of undefined
		at Blockly.FieldLabel.Blockly.Field.render_ (blockly_compressed.js:1360)
		at Blockly.FieldLabel.Blockly.Field.getSize (blockly_compressed.js:1363)
		at Blockly.BlockSvg.renderCompute_ (blockly_compressed.js:1488)
		at Blockly.BlockSvg.render (blockly_compressed.js:1483)
		at Blockly.BlockSvg.render (blockly_compressed.js:1483)
		at Blockly.BlockSvg.updateShape_ (blocks_sendto.js:235)
		at Blockly.BlockSvg.init (blocks_sendto.js:100)
		at Blockly.BlockSvg.Blockly.Block (blockly_compressed.js:1378)
		at new Blockly.BlockSvg (blockly_compressed.js:1437)
		at Blockly.WorkspaceSvg.newBlock (blockly_compressed.js:1284)

### Operating System and Browser

MacOS Chrome and Safari

**Comment by rachel-fenichel** [comment]
I can't reproduce with the code you've provided because it has some references that don't work (like Blockly.Words).

Please pare it down to a minimal repro case that I can play with in the playground.

**Comment by jogibear9988** [comment]
will do, hopefully tomorrow

**Comment by jogibear9988** [comment]
'use strict';

	goog.provide('Blockly.JavaScript.Sendto');

	goog.require('Blockly.JavaScript');

	Blockly.CustomBlocks = Blockly.CustomBlocks || [];
	Blockly.CustomBlocks.push('Sendto');

	Blockly.Sendto = {
		HUE: 310,
		blocks: {}
	};

	// --- sendTo Custom --------------------------------------------------
	Blockly.Sendto.blocks['sendto_custom'] =
		'<block type="sendto_custom">'
		+ '     <value name="INSTANCE">'
		+ '     </value>'
		+ '     <value name="COMMAND">'
		+ '     </value>'
		+ '     <value name="LOG">'
		+ '     </value>'
		+ '     <value name="WITH_STATEMENT">'
		+ '     </value>'
		+ '     <mutation with_statement="false" items="parameter1"></mutation>'
		+ '</block>';

	Blockly.Blocks['sendto_custom_container'] = {
		/**
		 * Mutator block for container.
		 * @this Blockly.Block
		 */
		init: function () {
			this.setColour(Blockly.Sendto.HUE);

			this.appendDummyInput()
				.appendField('sendto_custom_arguments');

			this.appendStatementInput('STACK');
			this.setTooltip('sendto_custom_arg_tooltip');
			this.contextMenu = false;
		}
	};

	Blockly.Blocks['sendto_custom_item'] = {
		/**
		 * Mutator block for add items.
		 * @this Blockly.Block
		 */
		init: function () {
			this.setColour(Blockly.Sendto.HUE);

			this.appendDummyInput('NAME')
				.appendField('sendto_custom_argument');

			this.setPreviousStatement(true);
			this.setNextStatement(true);
			this.setTooltip('sendto_custom_arg_tooltip');
			this.contextMenu = false;
		}
	};

	Blockly.Blocks['sendto_custom'] = {
		/**
		 * Block for creating a string made up of any number of elements of any type.
		 * @this Blockly.Block
		 */
		init: function () {
			var options = [];
			if (typeof main !== 'undefined' && main.instances) {
				for (var i = 0; i < main.instances.length; i++) {
					if (main.objects[main.instances[i]].common.messagebox) {
						var id = main.instances[i].substring('system.adapter.'.length);
						options.push([id, id]);
					}
				}
				if (!options.length) {
					options.push(['sendto_no_instances', '']);
				}
				/*for (var h = 0; h < scripts.hosts.length; h++) {
					options.push([scripts.hosts[h], scripts.hosts[h]]);
				}*/
				this.appendDummyInput('INSTANCE')
					.appendField('sendto_custom')
					.appendField(new Blockly.FieldDropdown(options), 'INSTANCE');
			} else {
				this.appendDummyInput('INSTANCE')
					.appendField('sendto_custom')
					.appendField(new Blockly.FieldTextInput('adapter.0'), 'INSTANCE');
			}

			this.appendDummyInput('COMMAND')
				.appendField('sendto_custom_command')
				.appendField(new Blockly.FieldTextInput('send'), 'COMMAND');

			this.setColour(Blockly.Sendto.HUE);

			this.itemCount_ = 1;
			this.updateShape_();
			this.setInputsInline(false);
			this.setPreviousStatement(true, null);
			this.setNextStatement(true, null);
			this.setMutator(new Blockly.Mutator(['sendto_custom_item']));
			this.setTooltip('sendto_custom_tooltip');
			this.setHelpUrl(getHelp('sendto_custom_help'));
		},
		/**
		 * Create XML to represent number of text inputs.
		 * @return {!Element} XML storage element.
		 * @this Blockly.Block
		 */
		mutationToDom: function () {
			var container = document.createElement('mutation');
			var names = [];
			for (var i = 0; i < this.itemCount_; i++) {
				var input = this.getInput('ARG' + i);
				names[i] = input.fieldRow[0].getValue();
			}

			container.setAttribute('items', names.join(','));
			container.setAttribute('with_statement', this.getFieldValue('WITH_STATEMENT') === 'TRUE');
			return container;
		},
		/**
		 * Parse XML to restore the text inputs.
		 * @param {!Element} xmlElement XML storage element.
		 * @this Blockly.Block
		 */
		domToMutation: function (xmlElement) {
			var names = xmlElement.getAttribute('items').split(',');
			this.itemCount_ = names.length;
			this.updateShape_(names, xmlElement.getAttribute('with_statement') == 'true');
		},
		/**
		 * Populate the mutator's dialog with this block's components.
		 * @param {!Blockly.Workspace} workspace Mutator's workspace.
		 * @return {!Blockly.Block} Root block in mutator.
		 * @this Blockly.Block
		 */
		decompose: function (workspace) {
			var containerBlock = workspace.newBlock('sendto_custom_container');
			containerBlock.initSvg();
			var connection = containerBlock.getInput('STACK').connection;
			for (var i = 0; i < this.itemCount_; i++) {
				var itemBlock = workspace.newBlock('sendto_custom_item');
				itemBlock.initSvg();
				connection.connect(itemBlock.previousConnection);
				connection = itemBlock.nextConnection;
			}
			return containerBlock;
		},
		/**
		 * Reconfigure this block based on the mutator dialog's components.
		 * @param {!Blockly.Block} containerBlock Root block in mutator.
		 * @this Blockly.Block
		 */
		compose: function (containerBlock) {
			var itemBlock = containerBlock.getInputTargetBlock('STACK');
			// Count number of inputs.
			var connections = [];
			var names = [];
			while (itemBlock) {
				connections.push(itemBlock.valueConnection_);
				itemBlock = itemBlock.nextConnection &&
					itemBlock.nextConnection.targetBlock();
			}
			// Disconnect any children that don't belong.
			for (var i = 0; i < this.itemCount_; i++) {
				var input = this.getInput('ARG' + i);
				var connection = input.connection.targetConnection;
				names[i] = input.fieldRow[0].getValue();
				if (connection && connections.indexOf(connection) === -1) {
					connection.disconnect();
				}
			}
			this.itemCount_ = connections.length;
			if (this.itemCount_ < 1) this.itemCount_ = 1;
			this.updateShape_(names);
			// Reconnect any child blocks.
			for (var j = 0; j < this.itemCount_; j++) {
				Blockly.Mutator.reconnect(connections[j], this, 'ARG' + j);

			}
		},
		getArgNames_: function () {
			var names = [];
			for (var n = 0; n < this.itemCount_; n++) {
				var input = this.getInput('ARG' + n);
				names.push(input.fieldRow[0].getValue());
			}
			return names;
		},
		/**
		 * Store pointers to any connected child blocks.
		 * @param {!Blockly.Block} containerBlock Root block in mutator.
		 * @this Blockly.Block
		 */
		saveConnections: function (containerBlock) {
			var itemBlock = containerBlock.getInputTargetBlock('STACK');
			var i = 0;
			while (itemBlock) {
				var input = this.getInput('ARG' + i);
				itemBlock.valueConnection_ = input && input.connection.targetConnection;
				itemBlock = itemBlock.nextConnection && itemBlock.nextConnection.targetBlock();
				i++;
			}
		},
		/**
		 * Modify this block to have the correct number of inputs.
		 * @private
		 * @this Blockly.Block
		 */
		updateShape_: function (names, withStatement) {
			if (this.getInput('LOG'))
				this.removeInput('LOG');
			if (this.getInput('WITH_STATEMENT'))
				this.removeInput('WITH_STATEMENT');
			names = names || [];
			var _input;
			// Add new inputs.
			for (var i = 0; i < this.itemCount_; i++) {
				_input = this.getInput('ARG' + i);
				if (!_input) {
					_input = this.appendValueInput('ARG' + i);
					if (!names[i]) names[i] = 'sendto_custom_argument' + (i + 1);
					_input.appendField(new Blockly.FieldTextInput(names[i]));

					var _shadow = this.workspace.newBlock('text');
					_shadow.setShadow(true);
					_shadow.outputConnection.connect(_input.connection);
					_shadow.initSvg();
					_shadow.render();
					//console.log('New ' + names[i]);
				} else {
					_input.fieldRow[0].setValue(names[i]);
					//console.log('Exist ' + names[i]);
					if (!_input.connection.isConnected()) {
						console.log('Create ' + names[i]);
						var shadow = this.workspace.newBlock('text');

						shadow.setShadow(true);

						shadow.outputConnection.connect(_input.connection);
						shadow.initSvg();
						shadow.render();
					}
				}
			}
			// Remove deleted inputs.
			var blocks = [];
			while (_input = this.getInput('ARG' + i)) {
				var b = _input.connection.targetBlock();
				if (b && b.isShadow()) {
					blocks.push(b);
				}
				this.removeInput('ARG' + i);
				i++;
			}
			if (blocks.length) {
				var ws = this.workspace;
				setTimeout(function () {
					for(var b = 0; b < blocks.length; b++) {
						ws.removeTopBlock(blocks[b]);
					}
				}, 100);
			}

			this.appendDummyInput('WITH_STATEMENT')
				.appendField('request_statement')
				.appendField(new Blockly.FieldCheckbox(withStatement ? 'TRUE': 'FALSE', function (option) {
					var withStatement = (option == true);
					this.sourceBlock_.updateShape_(this.sourceBlock_.getArgNames_(), withStatement);
				}), 'WITH_STATEMENT');

			this.appendDummyInput('LOG')
				.appendField('sendto_log')
				.appendField(new Blockly.FieldDropdown([
					['sendto_log_none',  ''],
					['sendto_log_info',  'log'],
					['sendto_log_debug', 'debug'],
					['sendto_log_warn',  'warn'],
					['sendto_log_error', 'error']
				]), 'LOG');

			// Add or remove a statement Input.
			var inputExists = this.getInput('STATEMENT');

			if (withStatement) {
				if (!inputExists) {
					this.appendStatementInput('STATEMENT');
				}
			} else if (inputExists) {
				this.removeInput('STATEMENT');
			}
		}
	};

	Blockly.JavaScript['sendto_custom'] = function (block) {
		var instance      = block.getFieldValue('INSTANCE');
		var logLevel      = block.getFieldValue('LOG');
		var command       = block.getFieldValue('COMMAND');
		var withStatement = block.getFieldValue('WITH_STATEMENT');
		var args = [];
		var logText;
		if (logLevel) {
			logText = 'console.' + logLevel + '("' + instance + ': " + "' + (args.length ? args.join(',') + '\n' : '') + '");\n'
		} else {
			logText = '';
		}
		var statement;
		if (withStatement === 'TRUE') {
			statement = Blockly.JavaScript.statementToCode(block, 'STATEMENT');
		}

		for (var n = 0; n < block.itemCount_; n++) {
			var input = this.getInput('ARG' + n);
			var val = Blockly.JavaScript.valueToCode(block, 'ARG' + n, Blockly.JavaScript.ORDER_COMMA);
			// if JSON
			if (val && val[0] === "'" && val[1] === '{') {
				val = val.substring(1, val.length - 1);
			}
			args.push('\n   "' + input.fieldRow[0].getValue() + '": ' + val);

			if (block.itemCount_ === 1 && !input.fieldRow[0].getValue()) {
				if (statement) {
					return 'sendTo("' + instance + '", "' + command + '", ' + val + ', function (result) {\n  ' + statement + '  });\n' + logText;
				} else {
					return 'sendTo("' + instance + '", "' + command + '", ' + val + ');\n' + logText;
				}
			}
		}

		if (statement) {
			return 'sendTo("' + instance + '", "' + command + '", {' + (args.length ? args.join(',') + '\n' : '') + '}, function (result) {\n  ' + statement + '  });\n' + logText;
		} else {
			return 'sendTo("' + instance + '", "' + command + '", {' + (args.length ? args.join(',') + '\n' : '') + '});\n' + logText;
		}
	};

**Comment by jogibear9988** [comment]
I've removed the words array access, now you should be able to reproduce

**Comment by jogibear9988** [comment]
with the version of blockly we use in ioBroker, this Code works (don't know wich block version exactly it is)

**Comment by rachel-fenichel** [comment]
Thanks for the updated code.  I have a fix for you, though it's not clear to me why this would be an issue now and not before.

The fix is to modify your `updateShape_` function.  Move the calls to `_shadow.outputConnections.connect(_input.connection)` to be after the calls to `shadow.render()` in both branches.

The issue is that when `render` is called on a child block, it is also called on all parent and ancestor blocks.  This propagates size changes up to every block that might be impacted by the child block changing.
It looks like you're calling render on the child block before the parent block has been initialized.  So when `render` is called on the parent block not all of the SVG elements that it expects exist.

```
if (!_input) {
  _input = this.appendValueInput('ARG' + i);
  if (!names[i]) names[i] = 'sendto_custom_argument' + (i + 1);
  _input.appendField(new Blockly.FieldTextInput(names[i]));

  var _shadow = this.workspace.newBlock('text');
  _shadow.setShadow(true);
  _shadow.outputConnection.connect(_input.connection);  <---- move this line
  _shadow.initSvg();
  _shadow.render();
  // move line to here
} else {
  _input.fieldRow[0].setValue(names[i]);
  if (!_input.connection.isConnected()) {
    console.log('Create ' + names[i]);
    var shadow = this.workspace.newBlock('text');

    shadow.setShadow(true);

    shadow.outputConnection.connect(_input.connection); <---- and this line
    shadow.initSvg();
    shadow.render();
    // and to here
  }
}
```

**Comment by jogibear9988** [comment]
I've updated the index.txt, I missed 2 lines.
I didn't see the error, cause ofg the error with the new blockly version.
I now copied over the one from iobroker where it works, then I saw that I forgot 2 lines.
So now the index.txt works with old blockly, with new it crashes

**Comment by rachel-fenichel** [comment]
@jogibear9988 yes, but I don't work on weekends.

For the `on_ext` block the problem is that you're calling `updateShape_` in your `init` function and it's relying on things that haven't been initialized.  The simple solution is to remove the call to `updateShape_` and update your toolbox XML to include a mutation tag, like so: `  <block type="on_ext"><mutation items="1"></mutation></block>`

That gets this block in the flyout:
![image](https://user-images.githubusercontent.com/13686399/51856932-d18c5180-22e5-11e9-9447-677ef51cfea6.png)

This works because it defers the call to `updateShape_` until after the block has been properly initialized, and `updateShape_` is still called from `domToMutation`.  

Again, I'm not sure why your code worked before, but I don't think we changed anything in the last release that should have had an impact here.

**Comment by jogibear9988** [comment]
<img width="595" alt="bildschirmfoto 2019-01-28 um 20 56 06" src="https://user-images.githubusercontent.com/364896/51862575-2c05cc00-233f-11e9-8d46-f7576086f8f5.png">

this works, but now you don't have the real block in the side bar...

**Comment by jogibear9988** [comment]
And there is also another issue, seems this is not one with our code.

Try to use the mutator and add another field, but not after the existing ones, no in front of them. Then you have broken shadow blocks on the designer

**Comment by rachel-fenichel** [comment]
It looks like you got rid of the call to `updateShape` but didn't add the mutation to the XML in the toolbox.  Adding the mutation to the XML solved both of those problems for me.

**Comment by jogibear9988** [comment]
you are right, the first poroblem got solved by this, but try my demo.  drag the element to designer, click on the mutator, drag another "on_ext_oid" in front of the other you will see that after that you hav broken shadows

**Comment by jogibear9988** [comment]
Is this a error in our code? Or will this also not work if we add shadows to original "list mutation block"?

**Comment by rachel-fenichel** [comment]
It's not an error in the list mutation block because that block doesn't have shadows.  When you add shadows, you're also responsible for managing them correctly.

**Comment by jogibear9988** [comment]
but the shadow block wich is too much is created from "respawnShadow" from within blockly

**Comment by jogibear9988** [comment]
so once I've attached a shadow, I'm not able to mange it any more by myself

**Comment by rachel-fenichel** [comment]
In this case you would need to remove the shadow dom from the connection before deleting your shadow block, then add it back later.  Dealing with details like this is part of digging deep with the mutator API.  This is not a bug in Blockly.
